browser/ui/tabs/split_view_browser_data.cc

```diff
@@ -107,4 +113,71 @@ void SplitViewBrowserData::RemoveObserver(
   observers_.RemoveObserver(observer);
 }
 
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure() = default;
+
+SplitViewBrowserData::OnTabDragEndedClosure::OnTabDragEndedClosure(
+    base::WeakPtr<SplitViewBrowserData> data,
+    base::OnceClosure closure)
+    : data_(data), closure_(std::move(closure)) {
+  CHECK(data_);
```

Comment:
why do you have a `CHECK` on a WeakPtr? That defeats the entire purpose of using a WeakPtr

Reply:
Good point. I think we even need to pass WeakPtr<SplitViewBrowserData>, here as we don't expect it to be invalidated during the closure's lifetime.